net/host.go

```diff
@@ -79,25 +85,37 @@ func (h *Host) P2pHost() P2pHost {
 	return h.h
 }
 
-// SetHandler sets the Handler instance used by the host.
-func (h *Host) SetHandler(handler Handler) {
+// SetHandlers sets the maker and taker instances used by the host.
+func (h *Host) SetHandlers(makerHandler MakerHandler, takerHandler TakerHandler) {
+	h.makerHandler = makerHandler
+	h.takerHandler = takerHandler
+
 	fn := func() bool {
-		return len(handler.GetOffers()) == 0
+		return len(makerHandler.GetOffers()) > 0 || h.isRelayer
 	}
-
-	h.handler = handler
 	h.h.SetShouldAdvertiseFunc(fn)
```

Comment:
I think we got the API on this wrong. Advertising always happens with the empty ("") namespace if nothing else. Instead of having the function ask if we should advertise, it should be asking *what* we should be advertising. That will have to be done in follow up work that updates the other repo though.